src/http.c

```diff
@@ -925,17 +983,71 @@ http_file_rewind(struct http_file *file)
 	file->offset = 0;
 }
 
-void
+size_t
+http_populate_cookies(struct http_request *req)
+{
+	int						 i, v;
+	size_t					 nlen;
+    struct http_cookie		*ck;
+    char					*c, *header, *value;
+    char					*cookies[HTTP_MAX_COOKIES];
+    const char				*semicol = ";";
+    const char				*comma = ",";
+    const char				*delim = NULL;
+
+    if (!http_request_header(req, "cookie", &c))
+	   return 0;
+
+    header = kore_strdup(c);
+    if (strstr(header, semicol) != NULL) {
+    	delim = semicol;
+    }
+    else if (strstr(header, comma) != NULL) {
+    	delim = comma;
+    }
+    else
+    	delim = semicol;
+
+    v = kore_split_string(header, delim, cookies, HTTP_MAX_COOKIES);
+    for (i = 0; i < v; i++) {
+    	for (c = cookies[i]; isspace(*c); c++)
+    	    ;
+
+    	if ( (value = strchr(c, '=')) == NULL) {
+	    	kore_log(LOG_ERR, "malformed cookie");
+	    	continue;
+	    }
+	    value++;
+
+	    nlen = value - c - 1;
+	    if (nlen > HTTP_MAX_COOKIENAME) {
+	    	kore_log(LOG_ERR, "cookie name is too long");
+	    	continue;
+	    }
+		ck = kore_pool_get(&http_cookie_pool);
+		ck->name = kore_malloc(sizeof(char) * (nlen + 1));
+		memset(ck->name, 0, sizeof(char) * (nlen + 1));
+		strncpy(ck->name, c, nlen);
```

Comment:
No.

Please don't ever do this.

You can easily replace the '=' character with a NUL-byte and just use kore_strdup() on it for the name instead of this error-prone idiom.